src/cow-react/modules/ethFlow/state/updater.tsx

```diff
@@ -0,0 +1,36 @@
+import { Percent } from '@uniswap/sdk-core'
+import { useEffect, useState } from 'react'
+import { useSetUserSlippageTolerance, useUserSlippageTolerance } from 'state/user/hooks'
+import { useIsEthFlow } from './hooks'
+
+export const ETH_FLOW_SLIPPAGE = new Percent(2, 100) // 2%
+
+export default function Updater() {
+  // use previous slippage for when user is not in native swap and set to native flow
+  const currentSlippage = useUserSlippageTolerance()
+
+  // save the last non eth-flow slippage amount to reset when user switches back to normal erc20 flow
+  const [erc20Slippage, setErc20Slippage] = useState<Percent | 'auto' | undefined>(currentSlippage)
+
+  const { isEthFlow } = useIsEthFlow()
+
+  const setUserSlippageTolerance = useSetUserSlippageTolerance()
+
+  // change slippage set to 2% when detected eth swap and option is set to use native flow
+  useEffect(() => {
+    if (isEthFlow) {
+      // user switched back to erc20, we need to reset their previous slippage back to what it was
```

Comment:
no it's correct although this section changed to be simpler. if the app detects user is currently in `isEthFlow` (user selects ETH) then we need to save what the slippage was previously (e.g when they had WETH selected w/0.23% slippage) so that we can set it back when user switches out of eth flow

Reply:
Right, I got that part.
But to me I interpret as this `if` is executed when the user switches away from `eth flow`, which is not the case.
It will only get into the conditional when `isEthFlow` is true, at which point we are not switching back to the previous slippage value.

Not saying the logic is wrong, just that the comment doesn't fit the code IMO